src/screens/VolumeCalibrationScreen.tsx

```diff
@@ -0,0 +1,231 @@
+import { useCallback, useEffect, useState, useRef } from 'react'
+import { Alert } from 'react-native'
+import { Audio } from 'expo-av'
+import { Entypo } from '@expo/vector-icons'
+import { Text, Box, Button, TrialRatingScale } from '@components'
+import { palette } from '@utils/theme'
+
+type VolumeCalibrationScreenParams = {
+  onFinishCalibration: (volume: number) => void
+}
+
+export const VolumeCalibrationScreen: React.FunctionComponent<VolumeCalibrationScreenParams> = ({
+  onFinishCalibration,
+}) => {
+  const volume = useRef(0.5)
+  const soundRef = useRef<Audio.Sound>()
+  const [stage, setStage] = useState(0)
+  const [countdown, setCountdown] = useState(3)
+  const [volumeRating, setVolumeRating] = useState(undefined)
+
+  const playStimuli = async () => {
+    // Set the volume
+    await soundRef.current?.setVolumeAsync(volume.current)
+    // Play the sound
+    await soundRef.current?.replayAsync()
+  }
+
+  useEffect(() => {
+    // Assign sound object to ref
+    Audio.Sound.createAsync(require('../assets/sounds/ding.wav'), {
+      shouldPlay: false,
+    }).then(({ sound }) => {
+      soundRef.current = sound
+    })
+  }, [])
+
+  const decreaseCountdown = (countdown: number, callback: () => void) => {
+    setTimeout(() => {
+      // If timer finished...
+      if (countdown == 0) {
+        callback()
+      } else {
+        setCountdown(countdown - 1)
+        return decreaseCountdown(countdown - 1, callback)
+      }
+    }, 1000)
+  }
+
+  const startCalibration = async () => {
+    // Show countdown timer
+    setStage(1)
+
+    // Trigger Countdown
+    const testCurrentVolume = () => {
+      setCountdown(3)
+      decreaseCountdown(countdown, async () => {
+        // Play Sound
+        await playStimuli()
+
+        // Ask participant what they thought
+        Alert.alert(
+          'How was that?',
+          'Is the volume of the sound uncomfortable?',
+          [
+            {
+              text: 'Yes',
+              onPress: () => {
+                // Switch to volume rating
+                setStage(2)
+              },
+            },
+            {
+              text: 'No',
+              onPress: () => {
+                // If reached max volume
+                if (volume.current >= 1.0) {
+                  setStage(2)
+                } else {
+                  // Increase volume and play again
+                  volume.current = +(volume.current + 0.1).toFixed(2)
```

Comment:
I think there might be a typo here. Not sure if you were going for `+=` or there's just an extra `+`.

Reply:
This is correct although it looks funky. It basically allows floating point addition without all the weirdness of floating point addition.